nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-nar-utils/src/main/java/org/apache/nifi/nar/NativeLibFinder.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.nar;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+
+public interface NativeLibFinder {
+    Logger LOGGER = LoggerFactory.getLogger(NativeLibFinder.class);
+
+    String OS = System.getProperty("os.name").toLowerCase();
+
+    List<File> getNativeLibDirs();
+
+    Map<String, Path> getNativeLibNameToPath();
+
+    String getTmpLibFilePrefix();
+
+    default Set<File> getUsrLibDirs() {
+        Set<File> usrLibDirs = Arrays.stream(getJavaLibraryPath().split(File.pathSeparator))
+                .map(String::trim)
+                .filter(pathAsString -> !pathAsString.isEmpty())
+                .map(File::new)
+                .map(toDir())
+                .filter(Objects::nonNull)
+                .collect(Collectors.toSet());
+
+        return usrLibDirs;
+    }
+
+    default String getJavaLibraryPath() {
+        return System.getProperty("java.library.path", "");
+    }
+
+    default Function<File, File> toDir() {
```

Comment:
This should probably not be defined as a Function but rather just a method that takes a File as an argument and returns a File. It can still be referenced as a function when appropriate.